### DISCUSSION TITLE
Duplicate ip prefix

### DISCUSSION THREAD

**Discussion by joyhu** [netbox-community/netbox#15388] [Help Wanted!] [answered]
![image](https://github.com/netbox-community/netbox/assets/608110/40bd847d-720e-45b9-bf15-cc54b69e6b17)
When I add an existing prefix and select a different tenant, the system shows that the newly added prefix and the Children prefix are the same as the original ones, and after clicking on the newly added prefix, I can also see the sub-prefixes of other tenants.

**Comment by candlerb** [comment, marked as answer]
That's correct.

If you want the IP prefixes to be distinct, you need to assign them to a different VRF (in Netbox, a "VRF" is effectively a "Network Namespace"). Therefore if tenants are using overlapping address space, you need to create a VRF for each tenant.

If you want to prevent inserting overlapping prefixes in the default (global) VRF, then set [`ENFORCE_GLOBAL_UNIQUE = True`](https://docs.netbox.dev/en/stable/configuration/miscellaneous/#enforce_global_unique) in configuration.py

**Reply by joyhu** [reply]
谢谢！

**Reply by joyhu** [reply]
Is it possible to restrict prefixes in VRf to be unique?

**Reply by candlerb** [reply]
Yes, this is set as an attribute on each individual VRF.
![image](https://github.com/netbox-community/netbox/assets/44789/54a8fc07-6671-434a-90b2-a42bb925f557)

**Reply by joyhu** [reply]
We are a company that does network maintenance for different customers. We hope to enter all customer data into netbox. Different customers are represented by tenants in netbox, but we hope that company employees can only see the data of authorized tenants when accessing netbox. . Is this possible in the current version? Thanks!
